src/popup-page/modules/GlobalPostageStamp.tsx

```diff
@@ -24,12 +24,11 @@ export default function GlobalPostageStamp(): ReactElement {
     if (globalState.globalPostageBatchEnabled) retrievePostageBatches()
   }, [globalState.globalPostageBatchEnabled])
 
-  const handleUseGlobalPostageBatch = (checked: boolean): void => {
-    dispatchGlobalState({ type: 'GLOBAL_POSTAGE_BATCH_ENABLED_SAVE', newValue: checked })
-
+  const handleUseGlobalPostageBatch = async (checked: boolean) => {
     if (checked) {
-      retrievePostageBatches()
+      await retrievePostageBatches()
```

Comment:
It needs postage batch IDs ready when flag is updated because of the code in bee-api.listener:

https://github.com/ethersphere/swarm-extension/blob/ee1a99c656eeec94e308ec820ebf1deed52ac6c1/src/background/listener/bee-api.listener.ts#L305-L309